domain/src/main/java/org/oppia/domain/audio/AudioPlayerController.kt

```diff
@@ -0,0 +1,112 @@
+package org.oppia.domain.audio
+
+import android.content.Context
+import android.media.MediaPlayer
+import android.net.Uri
+import android.provider.MediaStore
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import javax.inject.Inject
+import kotlin.concurrent.withLock
+
+/* AudioPlayerController controls interactions with an internal MediaPlayer,
+*  Use initializeMediaPlayer to set a source url for the audio
+*  Observe on getPlayState to get updates to status of MediaPlayer
+*  Use deinitializeMediaPlayer to release MediaPlayer and stop scheduling Seek Bar updates
+*/
+class AudioPlayerController @Inject constructor(val context: Context) {
+
+  inner class AudioMutableLiveData : MutableLiveData<PlayStatus>() {
+    override fun onActive() {
+      super.onActive()
+      seekBarActive = true
+    }
+
+    override fun onInactive() {
+      super.onInactive()
+      seekBarActive = false
+      stopUpdatingSeekBar()
+    }
+  }
+
+  enum class Status { PREPARED, COMPLETED, POSITION_UPDATE }
```

Comment:
Re: POSITION_UPDATE, we can just indicate that's a PLAYED state. Remember, we don't want to treat LiveData like an event delivery mechanism. LiveData updates can be missed, so we should always be reporting absolute state (e.g. that the audio is playing).